src/adapters/ethereum.ts

```diff
@@ -27,26 +32,44 @@ const getConnectedAccounts = (provider: Web3Provider): Promise<string[]> => {
     .then((accounts) => ethereumAccountsParser.parse(accounts));
 };
 
-interface IsPermitSupportedParams {
-  account: string;
+interface DiscoverPermitParams {
   chain: Chain;
   token: Token;
 }
 
-const isPermitSupported = async ({
-  account,
-  chain,
-  token,
-}: IsPermitSupportedParams): Promise<boolean> => {
+const discoverPermit = ({ chain, token }: DiscoverPermitParams): Promise<Permit> => {
```

Comment:
Renamed as `getPermit`.

I named the function `discoverPermit` because it indeed "discovers the `Permit`" doing up to 3 different calls. It's true though that this is an implementation detail.